test/tools/libtesteth/boostTest.cpp

```diff
@@ -90,37 +86,32 @@ void setDefaultOrCLocale()
 //Custom Boost Unit Test Main
 int main( int argc, char* argv[] )
 {
+	std::string const dynamicTestSuiteName = "RandomTestCreationSuite";
 	setDefaultOrCLocale();
 	try
 	{
 		//Initialize options
 		dev::test::Options const& opt = dev::test::Options::get(argc, argv);
 		if (opt.createRandomTest)
 		{
-			//disable initial output
+			// Disable initial output as the random test will output valid json to std
 			oldCoutStreamBuf = std::cout.rdbuf();
 			oldCerrStreamBuf = std::cerr.rdbuf();
 			std::cout.rdbuf(strCout.rdbuf());
 			std::cerr.rdbuf(strCout.rdbuf());
 
 			for (int i = 0; i < argc; i++)
 			{
-				std::string arg = std::string{argv[i]};
-
 				//replace test suite to random tests
+				std::string arg = std::string{argv[i]};
 				if (arg == "-t" && i+1 < argc)
-					argv[i+1] = (char*)std::string("RandomTestCreationSuite").c_str();
-
-				//don't pass long raw test input to boost
-				if (arg == "--checktest")
 				{
-					argc = i + 1;
+					argv[i+1] = (char*)dynamicTestSuiteName.c_str();
```

Comment:
Before doing this assignment, make sure `argc > i + 1` holds.

Reply:
this is done one line up

Reply:
OK.  The next question: why is it necessary to edit the user input?

Reply:
this is a trick to overwrite boost option -t GeneralStateTests     with   -t randomTestCreationSuite which is dynamicly generated.

Reply:
so you type  -t GeneralStateTests -- --createRandomTest  
but instead it runs boost with -t randomTestCreationSuite 
which then reads what particular test to generate. in this case it is GeneralStateTest

Reply:
Will you add a comment explaining these?